shell/common/engine.h

```diff
@@ -227,6 +227,11 @@ class Engine final : public RuntimeDelegate, PointerDataDispatcher::Delegate {
     ///                              collected and send back to Dart.
     ///
     virtual void SetNeedsReportTimings(bool needs_reporting) = 0;
+
+    //--------------------------------------------------------------------------
+    /// @brief      Ask the delegate if should have surface.
+    ///
+    virtual bool HaveSurface() = 0;
```

Comment:
elaborate the comment to answer:
1. When can something not have a surface? 
2. Where is this commonly used?

Also call this method `HasSurface`.